src/Template/Bake/config/snapshot.ctp

```diff
@@ -131,11 +136,16 @@ class <%= $name %> extends AbstractMigration
             ->addIndex(
                 [<%
                     echo $this->Migration->stringifyList($index['columns'], ['indent' => 5]);
-                %>]<% echo ($index['type'] === 'fulltext') ? ',' : ''; %>
-
-                <%- if ($index['type'] === 'fulltext'): %>
-                ['type' => 'fulltext']
-                <%- endif; %>
+                %>],
+                [<%
+                    $indexOptions = [
+                        'name' => $name
+                    ];
+                    if ($constraint['type'] === 'fulltext') {
```

Comment:
It would probably be better to use the `if ():` and `endif` way of creating if statements. http://php.net/manual/en/control-structures.alternative-syntax.php